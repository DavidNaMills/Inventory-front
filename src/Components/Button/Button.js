import React from 'react';
import classes from './button.module.scss';

const allTypes = {
    base: {
        base: classes.button__base,
        shadow: classes.button__base_shadow,
        border: classes.button__base_border,
        skeleton: classes.button__base_skeleton,
    },

    submit: {
        base: classes.button__submit,
        shadow: classes.button__submit_shadow,
        border: classes.button__submit_border,
        skeleton: classes.button__submit_skeleton,
    },

    clear: {
        base: classes.button__clear,
        shadow: classes.button__clear_shadow,
        border: classes.button__clear_border,
        skeleton: classes.button__clear_skeleton,
    },

    danger: {
        base: classes.button__danger,
        shadow: classes.button__danger_shadow,
        border: classes.button__danger_border,
        skeleton: classes.button__danger_skeleton,
    },

    information: {
        base: classes.button__information,
        shadow: classes.button__information_shadow,
        border: classes.button__information_border,
        skeleton: classes.button__information_skeleton,
    },
};


const Button = ({ config = {}, onClick, label = null, skeleton = false, raised = false, full = false, type = 'base', border = false, bold = false, wide = false, isDisabled = false, Icon = null, noStyle=null }) => {
    const fullStyle = [classes.button];
    let which = type;
    const style = {};

    if (isDisabled) {
        fullStyle.push(classes.button__disabled);
    } else if(noStyle) { fullStyle.push(classes.button_noStyle);
    } else {
        if (allTypes[type]) {
            which = type;
            fullStyle.push(allTypes[type].base);
        } else {
            which = 'base';
            fullStyle.push(allTypes['base'].base);
        }

        if (skeleton) { fullStyle.push(allTypes[which].skeleton); }
        if (raised) { fullStyle.push(allTypes[which].shadow); }
        if (border) { fullStyle.push(allTypes[which].border); }
    }
    if (full) { fullStyle.push(classes.button_full); }
    if (bold) { fullStyle.push(classes.button_bold); }
    
    
    if (wide) {
        style.width = '30%';
    }

    return (
        <button
            {...config}
            onClick={onClick}
            style={style}
            className={fullStyle.join(' ')}
            disabled={isDisabled}
        ><p>{label}</p>{Icon && <Icon />}</button>
    )
}

export default React.memo(Button);