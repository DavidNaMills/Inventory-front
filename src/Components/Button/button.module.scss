@import '../../styles/mixins/buttonMixins.module.scss';
@import '../../styles/variables/colors.module.scss';

.button{
    display: flex;
    align-content: center;
    justify-content: center;
    font-size: 16px;
    border: none;
    padding: 10px;
    padding-left: 20px;
    padding-right: 20px;
    border-radius: 2px;

    p{
        padding-top: 5px;
        padding-right: 10px;
    }

    &__disabled{
        background-color: $disabled-back;
        color: $disabled-color;
    }

    &__base{
        @include buttonStyle($base-back, $base-color);
        &:hover{
            @include onHover($base-shadow, $light-text, $base-back);
        }

        &_shadow{@include buttonShadow($base-shadow);}
        &_border{@include buildBorder($base-color);}
        &_skeleton{@include buildSkeleton($base-back, $base-color); }
    };
    
    &__submit{
        @include buttonStyle($submit-back, $submit-color);
        &:hover{
            @include onHover($submit-shadow, $light-text, $submit-back);
        }

        &_shadow{ @include buttonShadow($submit-shadow); }
        &_border{ @include buildBorder($submit-color); }
        &_skeleton{@include buildSkeleton($submit-back, $submit-color); }
    }
    
    &__clear{
        @include buttonStyle($clear-back, $clear-color);
        &:hover{
            @include onHover($clear-shadow, $clear-color, $clear-back);
        }

        &_shadow{ @include buttonShadow($clear-shadow); }
        &_border{ @include buildBorder($clear-color); }
        &_skeleton{@include buildSkeleton($clear-back, $clear-color); }
    }
    
    &__danger{
        @include buttonStyle($danger-back, $danger-color);
        &:hover{
            @include onHover($danger-shadow, $light-text, $danger-back);
        }
        &_shadow{ @include buttonShadow($danger-shadow); }
        &_border{ @include buildBorder($danger-color); }
        &_skeleton{@include buildSkeleton($danger-back, $danger-color); }
    }
    
    &__information{
        @include buttonStyle($information-back, $information-color);
        &:hover{
            @include onHover($information-shadow, $light-text, $information-back);
        }
        &_shadow{ @include buttonShadow($information-shadow); }
        &_border{ @include buildBorder($information-color); }
        &_skeleton{@include buildSkeleton($information-back, $information-color); }
    }

    &_flat{
        border: none;
        box-shadow: none;
        -webkit-box-shadow: none;
        -moz-box-shadow: none;
    }

    &_full{
        width: 100%;
    }

    &_bold{
        font-weight: bold;
    }

    &_noStyle{
        background-color: transparent;
        border: none;
        box-shadow: none;

        &:hover{
            background-color: rgb(212, 212, 212);
            padding-top: 10px;
            padding-bottom: 10px;
            padding-left: 5px;
            padding-right: 10px;
            border-radius: 100px;
        }
    }
}