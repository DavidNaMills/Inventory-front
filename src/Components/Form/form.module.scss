@import '../../styles/variables/sizing.module.scss';
@import '../../styles/variables/colors.module.scss';

@mixin form {
    -webkit-backdrop-filter: blur(5px);
    backdrop-filter: blur(5px);
    // background-color: rgba(242, 242, 242, 0.5);

    margin-bottom: $largeSpacing;
    margin-top: $mediumSpacing;
    z-index: 1;
}



.form{
    border-radius: 3px;
    padding: 15px;
    
    
    -webkit-box-shadow: $blur $shadow-blue;
    -moz-box-shadow: $blur $shadow-blue;
    box-shadow: $blur $shadow-blue;

    &__buttonSpacing{
        &_large{
            margin-top: $mediumSpacing;
        }
        &_medium{
            margin-top: $extraSmallSpacing;
        }
    }
}

.blur{
    &__normal{
        width: 40%;
        @include form;
    }
    
    &_mobile{
        width: 80%;
        @include form;
    }
}




// @import '../../styles/variables/sizing.module.scss';
// @import '../../styles/variables/colors.module.scss';

// @mixin form {
//     border-radius: 3px;
//     padding: 15px;
//     margin-bottom: $largeSpacing;
//     margin-top: $mediumSpacing;
    
//     -webkit-box-shadow: $blur $shadow-blue;
//     -moz-box-shadow: $blur $shadow-blue;
//     box-shadow: $blur $shadow-blue;


// }


// .form{
//     &__normal{
//         width: 40%;
//         @include form;
//     }
    
//     &_mobile{
//         width: 80%;
//         @include form;
//     }

//     &__buttonSpacing{
//         &_large{
//             margin-top: $mediumSpacing;
//         }
//         &_medium{
//             margin-top: $extraSmallSpacing;
//         }
//     }
// }

// .blur{
//     // width:100%;
//     -webkit-backdrop-filter: blur(10px);
//     backdrop-filter: blur(5px);
//     background-color: rgba(255, 255, 255, 0.5); 
// }