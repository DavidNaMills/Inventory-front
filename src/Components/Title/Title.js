import React from 'react';
import classes from './title.module.scss';


const Title = (props) => {
    const { size = null, message = '', center = false } = props;
    let element = null;

    switch (size) {
        case 'h1':
            element = <h1 className={[classes.title_h1, center ? classes.title__center : null].join(' ')} >{message}</h1>
            break;
        case 'h2':
            element = <h2 className={[classes.title_h2, center ? classes.title__center : null].join(' ')}>{message}</h2>
            break;
        case 'h3':
            element = <h3 className={[classes.title_h3, center ? classes.title__center : null].join(' ')}>{message}</h3>
            break;
        case 'h4':
            element = <h4 className={[classes.title_h4, center ? classes.title__center : null].join(' ')}>{message}</h4>
            break;
        case 'h5':
            element = <h5 className={[classes.title_h5, center ? classes.title__center : null].join(' ')}>{message}</h5>
            break;
        case 'h6':
            element = <h6 className={[classes.title_h6, center ? classes.title__center : null].join(' ')}>{message}</h6>
            break;
        default:
            element = <h2 className={[classes.title_h2, center ? classes.title__center : null].join(' ')}>{message}</h2>
            break;
    };

    return (
        <div>
            {element}
        </div>
    )
}

export default React.memo(Title);