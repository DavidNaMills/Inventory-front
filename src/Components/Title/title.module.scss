@import '../../styles/variables/colors.module.scss';
@import '../../styles/fonts/fontSizes.module.scss';
@import '../../styles/mixins/title.module.scss';

.title {
    @include setTitle($h2, $title-color);
    
    &_h1{
        @include setTitle($h1, $title-color);
    };

    &_h2{
        @include setTitle($h2, $title-color);
    };
    
    &_h3{
        @include setTitle($h3, $title-color);
    }

    &_h4{
        @include setTitle($h4, $title-color);
    }

    &_h5{
        @include setTitle($h5, $title-color);
    }

    &_h6{
        @include setTitle($h6, $title-color);
    }

    &__center{
        text-align: center;
    }


}