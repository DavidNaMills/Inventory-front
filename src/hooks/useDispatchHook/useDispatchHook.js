import { useDispatch } from 'react-redux';

import { userLogin, userLogout, changeCurrentLocation } from '../../store/userStore/userActions';
import { startInitTypes, addNewProductType, addNewLocation, removeLocationType } from '../../store/typesStore/typesStoreActions';
import {createMessage} from '../../store/messagingStore/messagingActions';

const useDispatchHook = () => {
    const dispatch = useDispatch();

    const userLoginDispatch = (userData) => dispatch(userLogin(userData));
    const userLogoutDispatch = () => dispatch(userLogout());
    const changeCurrentLocationDispatch = (id) => dispatch(changeCurrentLocation(id));

    const initTypesDispatch = () => dispatch(startInitTypes());
    const addNewProductDispatch = (newType) => dispatch(addNewProductType(newType));
    const addNewLocationDispatch = (newLocation) => dispatch(addNewLocation(newLocation));
    const removeLocationDispatch = (id) => dispatch(removeLocationType(id));

    const createMessageDispatch = (data) => dispatch(createMessage(data));

    return {
        userLoginDispatch,
        userLogoutDispatch,
        changeCurrentLocationDispatch,

        initTypesDispatch,
        addNewProductDispatch,
        addNewLocationDispatch,
        removeLocationDispatch,

        createMessageDispatch
    }
}

export default useDispatchHook;