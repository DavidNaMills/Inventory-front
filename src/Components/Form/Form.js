import React from 'react';
import classes from './form.module.scss';
import { isMobileOnly } from 'react-device-detect';
import { useLocale } from '../../Context/LocaleContext';
import FormFactory from './Factory/FormFactory';
import Button from '../Button/Button';
import Title from '../Title/Title';
import PublishIcon from '@material-ui/icons/Publish';
import ClearIcon from '@material-ui/icons/Clear';

import objToArray from '../../helpers/objToArray/objToArray';


const Form = (props) => {
    const locale = useLocale();
    const {
        displayFields,
        user,
        fullForm,
        controlForm,
        updating,
        withTitle = true,

        submitCb = null,
        clearCb = null
    } = props;
    // isMobileOnly ? classes.form_mobile : classes.form__normal}
    return (
        <div className={isMobileOnly ? classes.blur_mobile : classes.blur__normal}>
            <div className={classes.form}>
                {withTitle && <Title message={updating ? `${locale.updating} ${updating}` : locale.creating} size='h3' />}
                <form onSubmit={submitCb}>
                    {
                        objToArray(fullForm).map((x, y) =>
                            <FormFactory
                                key={y}
                                config={x}
                                role={user.role}
                                displayFields={displayFields}
                                onChange={controlForm}
                            />
                        )
                    }
                    <div className={classes.form__buttonSpacing_large}>
                        <Button config={{ type: 'submit' }} type="submit" label={updating ? locale.update : locale.submit} Icon={PublishIcon} full />
                    </div>
                </form>
                {clearCb && <div className={classes.form__buttonSpacing_medium}>
                    <Button onClick={clearCb} label={locale.clear} type='clear' Icon={ClearIcon} full />
                </div>}
            </div>
        </div>
    )
}

export default Form;