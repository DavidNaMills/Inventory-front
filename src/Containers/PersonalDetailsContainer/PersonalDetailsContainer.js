import React, { useState } from 'react';
import classes from './personal.module.scss';
import { useSelector } from 'react-redux';
import { personalFormConfig } from '../../configs/formConfigs/personalFormConfig';
// import TimedLogout from '../../Components/TimedLogout/TimedLogout';

import useForm from '../../hooks/useForm/useForm';
import useAxios from '../../hooks/useAxios/useAxios';
import useDispatchHook from '../../hooks/useDispatchHook/useDispatchHook';
import axios from '../../Axios/axiosInstance';
import { useLocale } from '../../Context/LocaleContext';
import { isMobileOnly } from 'react-device-detect';
import Form from '../../Components/Form/Form';
import Title from '../../Components/Title/Title';
import Button from '../../Components/Button/Button';
import ExitToAppIcon from '@material-ui/icons/ExitToApp';

import { localeOptions } from '../../Locale/localeOptions';
import useChangeLanguage from '../../hooks/useChangeLanguage/useChangeLanguage';


const FORM_NAME = 'personal';

const PersonalDetailsContainer = (props) => {
    const locale = useLocale();
    const { staff } = useSelector(state => state);
    const { makeChange } = useChangeLanguage();

    const displayFields = staff.config.formName[FORM_NAME].create;

    const { createMessageDispatch } = useDispatchHook();
    const {
        fullForm,
        buildFormObject,
        controlForm,
        clearForm,
        validatePartial
    } = useForm({ formConfig: personalFormConfig(locale.errors) });

    const { makeCall } = useAxios(axios);

    const submitForm = (e) => {
        e.preventDefault();

        const valid = validatePartial();
        if (valid) {
            const all = buildFormObject(true);
            delete all.passwordConfirm;

            makeCall({
                url: `/staff/${staff.staff._id}`,
                method: 'PUT',
                data: all,
                callBack: updateCallback
            });
        }
    }

    const updateCallback = (res) => {
        if (res.error) {

        } else {
            createMessageDispatch({
                message: locale.succ_personal_update,
                messageType: 'succType'
            });
            // resetUpdate();
        }
    }


    return (
        <div className={classes.personal}>
            <Title size={'h1'} message={locale.cont_personal} center />
            <div style={{ marginTop: '40px' }}>
                <Button
                    onClick={() => props.history.push('/logout')}
                    label={locale.logout}
                    type='information'
                    full={isMobileOnly ? true : false}
                    Icon={ExitToAppIcon}
                />
            </div>
            <Form
                formConfig={fullForm}
                displayFields={displayFields}
                controlForm={controlForm}
                title={locale[staff.config.formName[FORM_NAME].title]}
                formName={FORM_NAME}
                user={staff.config}
                fullForm={fullForm}
                submitCb={submitForm}
                clearCb={clearForm}
                updating={locale.personal_update}
            />
            <div className={isMobileOnly ? classes.blur_mobile : classes.blur__normal}>
                <div className={classes.form}>
                    <Title size={'h3'} message={locale['chng_lang']} />
                    <select onChange={e => makeChange(e.target.value)} className={classes.personal__selectStyle} value={locale.which_lang}>
                        {
                            localeOptions.map(x =>
                                <option key={x.name} value={x.value}>
                                    {x.name}
                                </option>
                            )
                        }
                    </select>
                </div>
            </div>
        </div>
    )
}

export default PersonalDetailsContainer;