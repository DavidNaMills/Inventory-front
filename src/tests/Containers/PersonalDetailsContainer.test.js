import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

import '../testHelpers/context';
import './containerHelper';

import PersonalDetailsContainer from '../../Containers/PersonalDetailsContainer/PersonalDetailsContainer';
import Form from '../../Components/Form/Form';
import Title from '../../Components/Title/Title';
import Button from '../../Components/Button/Button';
import ExitToAppIcon from '@material-ui/icons/ExitToApp';

jest.mock('../../hooks/useDispatchHook/useDispatchHook');
import useDispatchHook from '../../hooks/useDispatchHook/useDispatchHook';

const mockCreateMessageDispatch = jest.fn();


describe('<PersonalDetailsContainer/> test suite', ()=>{
    let wrapper;

    beforeEach(() => {
        useDispatchHook.mockImplementation(() => ({
            userLogoutDispatch: mockCreateMessageDispatch
        }));
        wrapper = shallow(<PersonalDetailsContainer/>);
    });

    it('matches snapshot', ()=>{
        const comp = renderer.create(<PersonalDetailsContainer/>);
        const app = comp.toJSON();
        expect(app.exists).toMatchSnapshot();
    });
    
    it('exists', ()=>{
        expect(wrapper.exists).toBeTruthy();
    });
    
    it('renders a <Title/> component', ()=>{
        const title = wrapper.find(Title);
        expect(title.length).toBe(1);
        expect(title.props()).toHaveProperty('size', 'h1');
        expect(title.props()).toHaveProperty('message', 'Personal');
        expect(title.props()).toHaveProperty('center', true);
    });
    
    it('renders a <Form/> component', ()=>{
        const form = wrapper.find(Form);
        expect(form.length).toBe(1);
        expect(form.props()).toHaveProperty('formName', 'personal');
        expect(form.props()).toHaveProperty('updating', 'details');
    });
    
    it('renders a <Button/> component', ()=>{
        const btn = wrapper.find(Button);
        expect(btn.length).toBe(1);
        expect(btn.props()).toHaveProperty('label', 'Logout');
        expect(btn.props()).toHaveProperty('type', 'information');
        expect(btn.props()).toHaveProperty('full', false);
        expect(btn.props()).toHaveProperty('Icon', ExitToAppIcon);
    });
});