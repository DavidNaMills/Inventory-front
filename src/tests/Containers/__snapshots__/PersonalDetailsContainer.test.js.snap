// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PersonalDetailsContainer/> test suite matches snapshot 1`] = `undefined`;
